src/Backend.elm

```diff
@@ -230,63 +235,108 @@ update msg model =
 updateFromFrontend : SessionId -> ClientId -> ToBackend -> BackendModel -> ( BackendModel, Cmd BackendMsg )
 updateFromFrontend sessionId clientId msg model =
     case msg of
-        SubmitFormRequest priceId a ->
+        SubmitFormRequest a ->
             case ( Untrusted.purchaseForm a, model.ticketsEnabled ) of
                 ( Just purchaseForm, TicketsEnabled ) ->
-                    case priceIdToProductId model priceId of
-                        Just productId ->
-                            let
-                                availability =
-                                    Inventory.slotsRemaining model
-
-                                validProductAndForm =
-                                    case ( productId == Id.fromString Product.ticket.couplesCamp, purchaseForm ) of
-                                        ( True, CouplesCampTicketPurchase _ ) ->
-                                            True && availability.couplesCampTicket
-
-                                        ( False, CampTicketPurchase _ ) ->
-                                            True && availability.campTicket
-
-                                        ( False, CampfireTicketPurchase _ ) ->
-                                            True && availability.campfireTicket
-
-                                        _ ->
-                                            False
-
-                                sponsorshipIdM =
-                                    purchaseForm |> PurchaseForm.commonPurchaseData |> .sponsorship
-
-                                sponsorship =
-                                    case sponsorshipIdM of
-                                        Just sponsorshipId ->
-                                            AssocList.get (Id.fromString sponsorshipId) model.prices
+                    let
+                        availability =
+                            Inventory.slotsRemaining model
+
+                        validProductAndForm =
+                            -- @TODO FIXME!
```

Comment:
I'm going to leave this for now, it's supposed to protect from people hand-crafting payloads from the frontend to skip form validation, but given the Lamdera wire format is opaque and undocumented, and there isn't really much to gain from someone going to the pains to reverse engineer it just to avoid filling in their full name or valid email, I don't think this is a realistic concern for our small event :)